File: openmp-wg-pII-2015/omp_imbalance.c
#include <math.h>
#include "ticktock.h"
int runloop(int i) {
    int sum=0,j;
    for (j=0;j<i;j++)
        sum+=i^j;
    return sum;
}

int main() {
    int i = 0;
    int N = 4e4;
    int arr[N];
    tick_tock tt;
    tick(&tt);
    #pragma omp parallel for schedule(dynamic) //schedule(static,100)
    for (i=0;i<N;i++) {
      arr[i]=runloop(i);
    }
    tock(&tt);
    return 0;
}

File: openmp-wg-pII-2015/omp_imbalance.f90

function runloop(i)
    integer :: i, my_sum,j
    my_sum = 0
    do j=1,i
        my_sum = my_sum + XOR(i,j)
    enddo
    runloop = my_sum

end function

program omp_imbalance
    integer,parameter :: N = 6e4
    integer :: i
    integer :: arr(N)
    call tick(it0,it1)
!$OMP PARALLEL DO schedule(static,200)
    do i=1,N
      arr(i)=runloop(i)
    enddo
!OMP END PARALLEL DO
    call tock(it0,it1);

end program

